.github/CONTRIBUTING.md

```diff
@@ -1,6 +1,130 @@
+
+When you file a pull request, you should add some tests to this file with this in mind -- for new features, try to cover possible use cases extensively (we use [Codecov](https://codecov.io/gh/Rdatatable/data.table) to make it a bit easier to see how well you've done to minimally cover any new code you've added); for bug fixes, include a minimal version of the problem you've identified and write a test to ensure that your fix indeed works, and thereby guarantee that your fix continues to work as the codebase is further modified in the future. We encourage you to scroll around in `tests.Rraw` a bit to get a feel for the types of examples that are being created, and how bugs are tested/features evaluated.
+
+What numbers should be used for new tests? Numbers should be new relative to current master at the time of your PR. If another PR is merged before yours, then there may be a conflict, but that is no problem, as [Matt will fix the test numbers when he merges your PR](https://github.com/Rdatatable/data.table/pull/4731#issuecomment-768858134).
+
+#### Using `test`
+
+See [`test` function manual](https://rdatatable.gitlab.io/data.table/reference/test.html).
+
+**References:** If you are not sure how to issue a PR, but would like to contribute, these links should help get you started:
+
+1. **[How to Github: Fork, Branch, Track, Squash and Pull request](https://gun.io/blog/how-to-github-fork-branch-and-pull-request/)**.
+2. **[Squashing Github pull requests into a single commit](http://eli.thegreenplace.net/2014/02/19/squashing-github-pull-requests-into-a-single-commit)**.
+3. **[Github help](https://help.github.com/articles/using-pull-requests/)** - you'll need the *fork and pull* model.
+
+Minimal first time PR
+---------------------
+
+```
+$ cd /tmp      # or anywhere safe to play
+$ git config --global core.autocrlf false   # Windows-only preserve \n in test data
+$ git clone https://github.com/Rdatatable/data.table.git
+$ cd data.table
+$ R CMD build .
+$ R CMD check data.table_1.11.5.tar.gz
```

Comment:
```suggestion
$ R CMD check data.table_1.15.99.tar.gz
```

Reply:
Maybe even more future proof?
```suggestion
$ R CMD check data.table_*.tar.gz
```

Reply:
This assumes there's no other tar.gz from other version laying around, unless we start with a clean-up step `rm data.table_*.tar.gz`.